jib-core/src/main/java/com/google/cloud/tools/jib/docker/ImageToTarballTranslator.java

```diff
@@ -57,19 +57,20 @@ public Blob toTarballBlob(ImageReference imageReference) throws IOException {
       Path layerContentFile = layer.getContentFile();
       String layerName = layerContentFile.getFileName().toString();
 
-      tarStreamBuilder.addEntry(new TarArchiveEntry(layerContentFile.toFile(), layerName));
+      tarStreamBuilder.addTarArchiveEntry(
+          new TarArchiveEntry(layerContentFile.toFile(), layerName));
       manifestTemplate.addLayerFile(layerName);
     }
 
     // Adds the container configuration to the tarball.
     Blob containerConfigurationBlob =
         new ImageToJsonTranslator(image).getContainerConfigurationBlob();
-    tarStreamBuilder.addEntry(
+    tarStreamBuilder.addTextEntryInUtf8(
         Blobs.writeToString(containerConfigurationBlob), CONTAINER_CONFIGURATION_JSON_FILE_NAME);
```

Comment:
So we're converting a blob to a string only for the tarStreamBuilder to write it back out as bytes? Really?

Reply:
I have to agree with you here. I can see the reason for wanting to make it a String parameter considering the use case, but the operation itself is so much more straightforward if we do everything with bytes.